src/ucp/wireup/wireup_cm.c

```diff
@@ -513,6 +582,14 @@ static unsigned ucp_cm_client_uct_connect_progress(void *arg)
     goto out;
 
 try_fallback:
+    if (status == UCS_ERR_BUFFER_TOO_SMALL) {
+        status = ucp_cm_ep_config_init_lanes(ep, &key,
+                ep_init_flags_prio[ucs_static_array_size(ep_init_flags_prio) - 1]);
+        if (status != UCS_OK) {
+            goto err;
+        }
+    }
```

Comment:
can we revert this to previous implementation?

Reply:
It may not.
1. In ```for-loop```, it won't get ep config and init lanes.
2. For the error ```UCS_ERR_BUFFER_TOO_SMALL```, it also needs to configure ep and init the lanes to reply pending request.
3. ```ucp_ep_set_failed``` needs to get the right failure reason.

Reply:
but we should be able to check address length right after ucp_cm_ep_client_initial_config_get - when we have the configuration key. so at this point we can do the fallback - before calling ucp_ep_realloc_lanes().
so I think the change in this for loop can be minimal